src/test/java/it/smartcommunitylab/aac/saml/SamlIdentityProviderConfigurationTest.java

```diff
@@ -0,0 +1,320 @@
+        Assertions.assertThat(rpRegistration.getAssertionConsumerServiceBinding()).isEqualTo(expRPRAssertionConsumerBinding);
+
+        RelyingPartyRegistration.AssertingPartyDetails partyDetails =  rpRegistration.getAssertingPartyDetails();
+        // sentinel value(s) used to check if auto configuration at least fetched data from the correct location
+        // This is a simplified approach. A deep and complete analysis will eventually be delegated to a second test
+        String expIdPEntityId = "https://idp.identityserver";
+        boolean expWantAuthNRequestSigned=true;
+        String expSSOLocation = "https://idp.identityserver/saml/sso";
+        Saml2MessageBinding expSSOBinding = Saml2MessageBinding.REDIRECT;
+
+        Assertions.assertThat(partyDetails.getEntityId()).isEqualTo(expIdPEntityId);
+        Assertions.assertThat(partyDetails.getSingleSignOnServiceLocation()).isEqualTo(expSSOLocation);
+        Assertions.assertThat(partyDetails.getSingleSignOnServiceBinding()).isEqualTo(expSSOBinding);
+        Assertions.assertThat(partyDetails.getWantAuthnRequestsSigned()).isEqualTo(expWantAuthNRequestSigned);
+    }
+
+    /**
+     * This test checks that a RelyingPartyRegistration contains information matching the saml configuration (with manual idp metadata discovery)
+     * @throws Exception
+     */
+    @Test
+    public void validateRelyingPartyRegistrationManual() throws Exception {
+
+        boolean automaticDiscoverySetting = false;
+        // Create (hardcoded) configuration map to be used for initialization
+        Map<String, Serializable> rawCfgMap = buildRawConfigurationMap(automaticDiscoverySetting);
+        SamlIdentityProviderConfigMap idProvCfgMap = new SamlIdentityProviderConfigMap();
+        idProvCfgMap.setConfiguration(rawCfgMap);
+
+        ConfigurableIdentityProvider cip = getDefaultConfigurableIdentityProvider();
+        IdentityProviderSettingsMap idProvSettingMap = getDefaultIdentityProviderSettingsMap();
+
+        SamlIdentityProviderConfig idProvCfg = new SamlIdentityProviderConfig(cip, idProvSettingMap, idProvCfgMap);
+
+        // check that the obtained saml Identity Provider matches the obtained map
+        RelyingPartyRegistration rpRegistration = idProvCfg.getRelyingPartyRegistration();
+        Assertions.assertThat(rpRegistration.getEntityId()).isEqualTo(expEntityID);
+        Assertions.assertThat(rpRegistration.getRegistrationId()).isEqualTo(expProviderId);
+        Assertions.assertThat(rpRegistration.getSigningX509Credentials().size()).isEqualTo(1);
+        Assertions.assertThat(rpRegistration.getAssertionConsumerServiceLocation()).isEqualTo(expRPRAssertionConsumerLocation);
+        Assertions.assertThat(rpRegistration.getAssertionConsumerServiceBinding()).isEqualTo(expRPRAssertionConsumerBinding);
+
+        RelyingPartyRegistration.AssertingPartyDetails partyDetails =  rpRegistration.getAssertingPartyDetails();
```

Comment:
partyDetails could be null. Assert that is not the case before accessing properties in it